sdk/daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/TypeDestructor.scala

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf
+package language
+
+import com.digitalasset.daml.lf.data.{Numeric, Ref}
+import com.digitalasset.daml.lf.language.{Util => AstUtil}
+
+import scala.collection.immutable.ArraySeq
+
+sealed abstract class TypeDestructor {
+
+  import TypeDestructor._
+
+  type Type <: Ast.Type
+
+  def wrap(typ: Ast.Type): Type
```

Comment:
Do we need `Type` and `wrap` at all? Did you add them in case we want to put some additional information inside `Type` in the future?